adapters/repos/db/index.go

```diff
@@ -811,8 +811,14 @@ func (i *Index) objectSearch(ctx context.Context, limit int, filters *filters.Lo
 		}
 	}
 
-	outObjects := make([]*storobj.Object, 0, len(shardNames)*limit)
-	outScores := make([]float32, 0, len(shardNames)*limit)
+	// use initially limited cap for reference (effectively unlimited) results
+	cap := int(config.DefaultQueryMaximumResults)
+	if !additional.ReferenceQuery {
+		cap = len(shardNames) * limit
```

Comment:
I think changing the limit value to `math.MaxInt` might cause a serious issue:
`math.MaxInt` is 2^63-1 on 64-bit architecture which is a huge number. 
Therefore, `len(shardNames) * limit` might result in an OOM exception or integer overflow.

Reply:
Would the allocation always be for `math.MaxInt`? that would indeed be a massive alloc each time?

Reply:
Let's assume that value of `limit` is math.MaxInt and !`additional.ReferenceQuery`
In this case `outObjects := make([]*storobj.Object, 0, cap)` will result into OOM or other nasty behaviour.

How can we make sure that the assumption is not true?

Reply:
Thx @redouan-rhazouani for bringing it up. Unfortunately there is a chance that nested reference query may result in huge number of matched element and we may end up with OOM issues. That solution is considered short-term as customers are waiting for it. We need to figure out how to handle nested/reference queries in smarter, more efficient way.

@etiennedi is right here, if method is called for reference (unlimited) query, initial allocation is done only for 10k results, if not then space for `len(shardNames) * limit` is allocated, as it was previously

Reply:
> How can we make sure that the assumption is not true?

@redouan-rhazouani we can not, limit is controlled by user and max allowed value is int64 (with default being 10k).
I can change initial size for both cases (nested / root query) to some lower fixed value. That should reduce the chance of OOM to scenarios with huge number of matches, comparing to current impl, where it is related to limit setting, not matches amount